src/main/java/net/mcreator/generator/GeneratorTokens.java

```diff
@@ -64,21 +65,47 @@ static String replaceVariableTokens(Object element, String rawname) {
 				String match = m.group(1);
 				Object value = null;
 				if (element != null) {
-					if (match.contains("()")) {
-						try {
-							value = element.getClass().getMethod(match.replace("()", "").trim()).invoke(element);
-						} catch (Exception e) {
-							LOG.warn("Failed to load token value " + match, e);
-						}
-					} else {
+					try {
+						value = match.contains("()") ?
+								element.getClass().getMethod(match.replace("()", "").trim()).invoke(element) :
+								element.getClass().getField(match.trim()).get(element);
+					} catch (Exception e) {
+						LOG.warn("Failed to load token value " + match, e);
+					}
+				}
+				rawname = rawname.replace("@[" + match + "]", String.valueOf(value));
+			}
+		}
+		return rawname;
+	}
+
+	static String replaceVariableTokens(GeneratableElement element, Object listItem, String rawname) {

```

Comment:
`replaceVariableTokens(Object, String)` could probably call `replaceVariableTokens(Object, (Object) null, String)` to not duplicate code

Reply:
Anything else? ~So that I can fit into 100 commits.~

Reply:
I prefer each change to have own commit as it is much easier to track flow of your changes by checking separate commits. Sometimes I mark file as reviewed and by splitting commits, I can only review changes, not the whole file again.

Reply:
✔️ Fixed